inference-engine/tests/functional/inference_engine/keep_constant_inputs_tests.cpp

```diff
@@ -0,0 +1,156 @@
+                   std::dynamic_pointer_cast<const ::ngraph::opset3::ShuffleChannels>(node);
+        };
+        auto nGraphFunc = clonedNetwork->getFunction();
+        // Disable shape inference (WA for generic operations)
+        ::ngraph::op::GenericIE::DisableReshape noReshape(nGraphFunc);
+
+        // Note: instead of running all Conversion Transformations you can make up your own transformation pipeline
+        ngraph::pass::CommonOptimizations(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet3ToOpSet2(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet2ToOpSet1(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet1ToLegacy(transformations_callback).run_on_function(nGraphFunc);
+        clonedNetwork = InferenceEngine::details::convertFunctionToICNNNetwork(nGraphFunc, *clonedNetwork, keep_constant_inputs);
+    }
+}
+
+TEST(KeepConstantInputsTests, ConvertConvolutionPoolReluNetworkWithTrue) {
+    std::shared_ptr <ngraph::Function> f_ptr;
+    f_ptr = ngraph::builder::subgraph::makeConvPoolRelu();
+    InferenceEngine::CNNNetwork originalNetwork(f_ptr);
+    std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork = InferenceEngine::cloneNetwork(originalNetwork);
+    transformNetwork(clonedNetwork, true);
+    InferenceEngine::CNNNetwork convertedNetwork(clonedNetwork);
+    std::cout << "Check for conversion of ConvolutionPoolRelu Network with keep_constant_inputs = true" << std::endl;
+    ASSERT_GT(numberOfInputsForLayerInCNNNetwork(convertedNetwork, "Convolution"), 1);
+}
+
+TEST(KeepConstantInputsTests, ConvertConvolutionPoolReluNetworkWithFalse) {
+    std::shared_ptr <ngraph::Function> f_ptr;
+    f_ptr = ngraph::builder::subgraph::makeConvPoolRelu();
+    InferenceEngine::CNNNetwork originalNetwork(f_ptr);
+    std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork = InferenceEngine::cloneNetwork(originalNetwork);
+    transformNetwork(clonedNetwork, false);
+    InferenceEngine::CNNNetwork convertedNetwork(clonedNetwork);
+    std::cout << "Check for conversion of ConvolutionPoolRelu Network with keep_constant_inputs = false" << std::endl;
+    ASSERT_EQ(numberOfInputsForLayerInCNNNetwork(convertedNetwork, "Convolution"), 1);
+}
+
+TEST(KeepConstantInputsTests, ConvertConvolutionBiasNetworkWithTrue) {
+    std::shared_ptr <ngraph::Function> f_ptr;
+    f_ptr = ngraph::builder::subgraph::makeConvBias();
+    InferenceEngine::CNNNetwork originalNetwork(f_ptr);
+    std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork = InferenceEngine::cloneNetwork(originalNetwork);
+    transformNetwork(clonedNetwork, true);
+    InferenceEngine::CNNNetwork convertedNetwork(clonedNetwork);
+    std::cout << "Check for conversion of ConvolutionBias Network with keep_constant_inputs = true" << std::endl;
+    ASSERT_GT(numberOfInputsForLayerInCNNNetwork(convertedNetwork, "Convolution"), 1);
```

Comment:
Please use accurate values.
I guess the number of inputs should be equal 3 if you keep constants for convolution with biases.

Reply:
For some networks in tests this number is 2.

Reply:
If you have conv+bias  network it always should be equal 3. Please use `ASSERT_EQ` with accurate values instead of `ASSERT_GT` or provide a proof why it is not possible.

Reply:
Done.